web/test/pages/onboarding/onboarding-foreign.test.tsx

```diff
@@ -177,6 +182,92 @@ describe("onboarding - foreign business", () => {
         screen.queryByText(Config.profileDefaults.FOREIGN.foreignBusinessType.errorTextRequired)
       ).not.toBeInTheDocument();
     });
+
+    it("deselects every other option if none is selected", async () => {
+      // GIVEN the none of the above option exists in the config file
+      const { page } = renderPage({ userData });
+      page.checkByLabelText(Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.employeesInNJ);
+      page.checkByLabelText(
+        Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.transactionsInNJ
+      );
+      page.checkByLabelText(Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.revenueInNJ);
+      page.checkByLabelText(Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.none);
+
+      expect(
+        (
+          screen.getByLabelText(
+            Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.none
+          ) as HTMLInputElement
+        ).checked
+      ).toEqual(true);
+      expect(
+        (
+          screen.getByLabelText(
+            Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.employeesInNJ
+          ) as HTMLInputElement
+        ).checked
+      ).toEqual(false);
+      expect(
+        (
+          screen.getByLabelText(
+            Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.transactionsInNJ
+          ) as HTMLInputElement
+        ).checked
+      ).toEqual(false);
+      expect(
+        (
+          screen.getByLabelText(
+            Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.revenueInNJ
+          ) as HTMLInputElement
+        ).checked
+      ).toEqual(false);
+
+      act(() => page.clickNext());
+      await waitFor(() => {
+        expect(userDataWasNotUpdated).toBeTruthy();
+      });
+    });
+
+    it("deselects none of the above when user selects a different option after selecting none of the above", async () => {
+      const { page } = renderPage({ userData });
+
+      page.checkByLabelText(Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.employeesInNJ);
+      page.checkByLabelText(Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.none);
+      page.checkByLabelText(
+        Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.transactionsInNJ
+      );
+
+      expect(
+        (
+          screen.getByLabelText(
+            Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.transactionsInNJ
+          ) as HTMLInputElement
+        ).checked
+      ).toEqual(true);
+      expect(
+        (
+          screen.getByLabelText(
+            Config.profileDefaults.FOREIGN.foreignBusinessType.optionContent.employeesInNJ
+          ) as HTMLInputElement
+        ).checked
+      ).toEqual(false);
```

Comment:
Having an assertion about `employeesInNJ` feels like a non-sequitur in this test. In your "when" action section, you check the `none` checkbox and the `transactionsInNJ` checkbox, so it makes sense to see assertions regarding their state. Is there a reason for asserting on a seemingly unrelated checkbox state?

Reply:
updated